Dockerfile

```diff
@@ -20,16 +11,49 @@ RUN apk add --update --no-cache \
     giflib-dev \
     pango-dev
 
-# Install dependencies
+WORKDIR /src
+
 COPY package*.json ./
-RUN npm install
+RUN npm ci --no-audit
+
+# -- CHECK STAGE --------------------------------
+
+FROM base AS check
+
+ARG CI
+ENV CI $CI
 
-# Copy source files
 COPY . .
+RUN npm run lint
+RUN npm test
+
+# -- BUILD STAGE --------------------------------
+
+FROM base AS build
 
-# Build project
+# Define build args & map them to environment variables
+ARG GA_TRACKING_ID
+ENV GA_TRACKING_ID $GA_TRACKING_ID
+
+# Build the project and then dispose files not necessary to run the project
+# This will make the runtime image as small as possible
+COPY . .
+RUN npx next telemetry disable > /dev/null
```

Comment:
`npx` runs the locally installed packages (in local `node_modules`) before going into the global path.

> Executes <command> either from a local node_modules/.bin, or from a central cache, installing any packages needed in order for <command> to run.

> By default, npx will check whether <command> exists in $PATH, or in the local project binaries, and execute that. If <command> is not found, it will be installed prior to execution.

Reply:
👍